src/Randomizer.SMZ3/Generation/Smz3Plandomizer.cs

```diff
@@ -45,10 +45,20 @@ public SeedData GenerateSeed(Config config, CancellationToken cancellationToken
                 playthrough = new Playthrough(config, Enumerable.Empty<Playthrough.Sphere>());
             }
 
+            var plandoName = config.PlandoConfig?.FileName ?? "unknown";
+            if (plandoName.StartsWith("Spoiler_Plando_"))
+            {
+                plandoName = plandoName.Replace("Spoiler_Plando_", "");
+                if (plandoName.Contains("_") && int.TryParse(plandoName.Substring(plandoName.IndexOf("_")+1), out _))
```

Comment:
I'm not entirely sure what this bit is doing, maybe a regexp would be more readable here?